packages/language-server/src/server.ts

```diff
@@ -25,15 +22,33 @@ import {
   hasSettingsForDocument,
   getSettingsForDocument,
 } from "@ui5-language-assistant/settings";
+import { ServerInitializationOptions } from "../api";
+import { getSemanticModel } from "./ui5-model";
+import { getCompletionItems } from "./completion-items";
+import { getXMLViewDiagnostics } from "./xml-view-diagnostics";
+import { getHoverResponse } from "./hover";
+import {
+  getFlexEnabledFlagForXMLFile,
+  isManifestDoc,
+  initializeManifestData,
+  updateManifestData,
+} from "./manifest-handling";
 
 const connection = createConnection(ProposedFeatures.all);
 const documents = new TextDocuments(TextDocument);
-let getSemanticModelPromise: Promise<UI5SemanticModel> | undefined = undefined;
+let semanticModelLoaded: Promise<UI5SemanticModel> | undefined = undefined;
+let manifestStateInitialized: Promise<void[]> | undefined = undefined;
 let initializationOptions: ServerInitializationOptions | undefined;
 let hasConfigurationCapability = false;
 
 connection.onInitialize((params: InitializeParams) => {
   const capabilities = params.capabilities;
+  const workspaceFolderUri = params.rootUri;
+  if (workspaceFolderUri !== null) {
```

Comment:
ensure that starting with empty workspace and then opening a new one/folder
will re-trigger this code.